membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -258,30 +266,36 @@ class AccountController(
           subscription <- SimpleEitherT(
             services.subscriptionService
               .current(contact)
-              .map(subs => subscriptionSelector(subscriptionName, s"the sfUser $contact", subs)),
+              .map(subs => subscriptionSelector(memsub.Subscription.Name(subscriptionName), s"the sfUser $contact", subs)),
           )
-          contributionPlan <- SimpleEitherT.fromEither(subscription.plan match {
-            case p if p.product == Contribution => Right(p)
+          catalog <- SimpleEitherT.rightT(services.futureCatalog)
+          contributionPlan = subscription.plan(catalog)
+          _ <- SimpleEitherT.fromEither(contributionPlan.product(catalog) match {
+            case Contribution => Right(())
             case nc => Left(s"$subscriptionName plan is not a contribution: " + nc)
           })
-          billingPeriod = contributionPlan.charges.billingPeriod.asInstanceOf[RecurringPeriod]
-          applyFromDate = contributionPlan.chargedThrough.getOrElse(contributionPlan.start)
-          currency = contributionPlan.charges.price.prices.head.currency
+          billingPeriod <- SimpleEitherT.fromEither(contributionPlan.billingPeriod.toEither)
+          recurringPeriod <- SimpleEitherT.fromEither(billingPeriod match {
+            case period: RecurringPeriod => Right(period)
+            case period: BillingPeriod.OneOffPeriod => Left(s"period $period was not recurring for contribution update")
+          })
+          applyFromDate = contributionPlan.chargedThroughDate.getOrElse(contributionPlan.start)
+          currency = contributionPlan.chargesPrice.prices.head.currency
           currencyGlyph = currency.glyph
-          oldPrice = contributionPlan.charges.price.prices.head.amount
+          oldPrice = contributionPlan.chargesPrice.prices.head.amount
```

Comment:
chargesPrice has the logic that was previously in the high level deserialiser (that returns the total pricing summary for all charges in the plan)
The reason it's not kept the old name is that charges is now called ratePlanCharges (so it would be `ratePlanCharges.prices`) and is now a list rather than a high level ChargeList so we couldn't add a dummy prices method without using extension methods.